### PR TITLE
Task #48987 [User] View Room

### PR DESCRIPTION
**Pull request by anhlh-3049**
## Related Tickets
- https://edu-redmine.sun-asterisk.vn/issues/48987

## WHAT (optional)
- View Room
## Evidence (Screenshot or Video)
Screenshot_15

### CODE REVIEW

**Comment by anhlh-3049** [comment]
ready

File: app/views/static_pages/index.html.erb [hunk 1]
```diff
@@ -184,244 +184,55 @@
...
-              </div>
-              <div class="col-md-3">
-                  <div class="room-rate">
-                      <h3>From</h3>
-                      <h1>$180</h1>
-                      <a href="#">Book Now</a>
-                  </div>
-              </div>
+            <div class="col-md-3">
+                <div class="room-img">
```
Comment (ghost) [inline app/views/static_pages/index.html.erb]: indent

File: app/views/static_pages/index.html.erb [hunk 2]
```diff
@@ -184,244 +184,55 @@
...
-                      <h1>$220</h1>
-                      <a href="#">Book Now</a>
-                  </div>
-              </div>
-          </div>
-          <hr>
-      </div>
-
-      <div class="col-md-12">
+      <% @rooms.each do |room| %>
```
Comment (ghost) [inline app/views/static_pages/index.html.erb]: dùng partial

File: app/controllers/room_views_controller.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+class RoomViewsController < ApplicationController
```
Comment (ghost) [inline app/controllers/room_views_controller.rb]: bỏ file này đi anh

**Review by ghost — COMMENTED** [review]
cmt

**Comment by anhlh-3049** [comment]
ready

File: app/controllers/rooms_controller.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  def index
+    @room_views = RoomView.all
+    @rooms = Room.all
+  end
+
+  private
+
+  def find_room
+    @room = Room.find_by(id: params[:id])
+    flash[:danger] = t ".flash_error"
```
Comment (ngocvt-0484) [inline app/controllers/rooms_controller.rb]: ?

File: app/controllers/rooms_controller.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class RoomsController < ApplicationController
+  def index
+    @room_views = RoomView.all
+    @rooms = Room.all
```
Comment (ngocvt-0484) [inline app/controllers/rooms_controller.rb]: k phân trang à

File: app/views/rooms/_room.html.erb [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+              <li><a href="#" data-toggle="modal" data-target="#modal-id-<%= room.id %>"><i class="fa fa-link"></i></a></li>
+            </ul>
+          </div>
+        </div>
+      </div>
+    </div>
+    <div class="col-md-6">
+        <div class="room-des">
+          <h3><%= link_to room.name, room %></h3>
+          <p><%= room.room_view.name %></p>
```
Comment (ngocvt-0484) [inline app/views/rooms/_room.html.erb:19]: dùng delegate

File: app/views/rooms/_room.html.erb [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            <li class="icon-8"></li>
+            <li class="icon-9"></li>
+            <li class="icon-10"></li>
+          </ul>
+        </div>
+    </div>
+    <div class="col-md-3">
+      <div class="room-rate">
+        <h3>From</h3>
+        <h1>$<%= room.price %></h1>
```
Comment (ngocvt-0484) [inline app/views/rooms/_room.html.erb]: giá tiền format tiền

File: app/views/rooms/_room.html.erb [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+      </div>
+    </div>
+  </div>
+  <hr>
+</div>
+<div class="modal fade" id="modal-id-<%= room.id %>" tabindex="-1" role="dialog" aria-hidden="true">
+  <div class="modal-dialog modal-dialog-centered" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h5 class="modal-title" id="exampleModalLongTitle">Image Room</h5>
```
Comment (ngocvt-0484) [inline app/views/rooms/_room.html.erb]: i18n

File: app/views/rooms/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+<div id="rooms">
+    <div class="container">
+        <div class="section-header">
+            <h2>Our Rooms</h2>
+            <p>
+                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas in mi libero. Quisque convallis, enim at venenatis tincidunt.
+            </p>
+        </div>
+        <div class="row">
+          <%= render :partial => "room", :collection => @rooms %>
```
Comment (ngocvt-0484) [inline app/views/rooms/index.html.erb]: render @rooms

File: app/views/rooms/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+                      <div class="input-group-append" data-target="#date-3" data-toggle="datetimepicker">
+                          <div class="input-group-text"><i class="fa fa-calendar"></i></div>
+                      </div>
+                  </div>
+              </div>
+          </div>
+          <div class="control-group col-md-3">
+              <label>Check-Out</label>
+              <div class="form-group">
+                  <div class="input-group date" id="date-4" data-target-input="nearest">
```
Comment (ngocvt-0484) [inline app/views/rooms/index.html.erb]: sai indent

**Comment by anhlh-3049** [comment]
ready

File: app/views/rooms/_room.html.erb [hunk 11]
```diff
@@ -0,0 +1,65 @@
+<div class="col-md-12">
+  <div class="row">
+    <div class="col-md-3">
+      <div class="room-img">
+        <div class="box12">
+          <%=  image_tag room.images.first&.display_image if room.images.first%>
```
Comment (ghost) [inline app/views/rooms/_room.html.erb]:
```suggestion
          <%=  image_tag room.images.first&.display_image if room.images.first %>
```

**Pull request merged**